satpy/writers/__init__.py

```diff
@@ -13,916 +13,50 @@
-    config_files = config_search_paths(os.path.join("writers", config_fn))
-    writer_kwargs.setdefault("config_files", config_files)
-    if not writer_kwargs["config_files"]:
-        raise ValueError("Unknown writer '{}'".format(writer))
-
-    try:
-        return load_writer_configs(writer_kwargs["config_files"],
-                                   **writer_kwargs)
-    except ValueError:
-        raise ValueError("Writer '{}' does not exist or could not be "
-                         "loaded".format(writer))
-
-
-def configs_for_writer(writer=None):
-    """Generate writer configuration files for one or more writers.
-
-    Args:
-        writer (Optional[str]): Yield configs only for this writer
-
-    Returns: Generator of lists of configuration files
-
-    """
-    if writer is not None:
-        if not isinstance(writer, (list, tuple)):
-            writer = [writer]
-        # given a config filename or writer name
-        config_files = [w if w.endswith(".yaml") else w + ".yaml" for w in writer]
-    else:
-        paths = get_entry_points_config_dirs("satpy.writers")
-        writer_configs = glob_config(os.path.join("writers", "*.yaml"), search_dirs=paths)
-        config_files = set(writer_configs)
-
-    for config_file in config_files:
-        config_basename = os.path.basename(config_file)
-        paths = get_entry_points_config_dirs("satpy.writers")
-        writer_configs = config_search_paths(
-            os.path.join("writers", config_basename),
-            search_dirs=paths,
-        )
-
-        if not writer_configs:
-            LOG.warning("No writer configs found for '%s'", writer)
-            continue
-
-        yield writer_configs
-
-
-def available_writers(as_dict=False):
-    """Available writers based on current configuration.
-
-    Args:
-        as_dict (bool): Optionally return writer information as a dictionary.
-                        Default: False
-
-    Returns: List of available writer names. If `as_dict` is `True` then
-             a list of dictionaries including additionally writer information
-             is returned.
-
-    """
-    writers = []
-    for writer_configs in configs_for_writer():
-        try:
-            writer_info = read_writer_config(writer_configs)
-        except (KeyError, IOError, yaml.YAMLError):
-            LOG.warning("Could not import writer config from: %s", writer_configs)
-            LOG.debug("Error loading YAML", exc_info=True)
-            continue
-        writers.append(writer_info if as_dict else writer_info["name"])
-    return writers
-
-
-def _determine_mode(dataset):
-    if "mode" in dataset.attrs:
-        return dataset.attrs["mode"]
-
-    if dataset.ndim == 2:
-        return "L"
-    if dataset.shape[0] == 2:
-        return "LA"
-    if dataset.shape[0] == 3:
-        return "RGB"
-    if dataset.shape[0] == 4:
-        return "RGBA"
-    raise RuntimeError("Can't determine 'mode' of dataset: %s" %
-                       str(dataset))
```

Comment:
FYI this function, as far as I can tell, isn't used anywhere so I deleted it (I did not move it to a submodule).